neo4j/driver_with_context.go

```diff
@@ -672,6 +672,13 @@ func ExecuteQueryWithTransactionConfig(configurers ...func(*TransactionConfig))
 	}
 }
 
+// ExecuteQueryWithAuthToken configures DriverWithContext.ExecuteQuery to overwrite the AuthToken for the session.
+func ExecuteQueryWithAuthToken(auth AuthToken) ExecuteQueryConfigurationOption {
```

Comment:
This is up for discussion:
Why name it `ExecuteQueryWithAuthToken` over `ExecuteQueryWithAuth`, given that the field on `ExecuteQueryConfiguration` (as well as the field on `SessionConfig`) is called `Auth`, not `AuthToken`?

Reply:
My thinking was that it's because you use an `AuthToken` to authenticate (like you do in `VerifyAuthentication(ctx context.Context, auth *AuthToken...)` and not an `auth.TokenManager` (like you do in `NewDriverWithContext(target string, auth auth.TokenManager...`) to avoid any confusion for the user.

We seem to call both types `auth` internally:
driverWithContext struct:
`auth auth.TokenManager`

SessionConfig:
`Auth *AuthToken`

To keep the API consistent thought I'm happy to rename it just to auth.

Reply:
Good argument you've got there :thinking:
I assume the fact that `TokenManager` was an afterthought already blurs the lines here as session config doesn't accept a manager, but the field is still called `Auth`.

Yeah, I don't know :upside_down_face: I'll leave it up to you to how to name this method.